php/libraries/Database.class.inc

```diff
@@ -1050,6 +1050,36 @@ class Database
             "Database class no longer extends PEAR"
         );
     }
+    /**
+     * Checks to see if a table exists in the Database
+     *
+     * @param string $test_name The table to check for.
+     *
+     * @return boolean true if the table exists
+     */
+    function tableExists($test_name)
+    {
+        $config   =& NDB_Config::singleton();
+        $database = $config->getSetting('database');
+        $query    = "SELECT table_name
+            FROM information_schema.tables
+            WHERE TABLE_SCHEMA=:db
+            AND table_name=:tn";
+
+        $DB     =& Database::singleton();
+        $result = $DB->pselectOne(
```

Comment:
This is a non-static function in the Database class. It should be using $this-> instead of getting another reference with a singleton call.